tfchain_bridge/pkg/bridge/withdraw.go

```diff
@@ -0,0 +1,133 @@
+package bridge
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/substrate-client"
+	subpkg "github.com/threefoldtech/tfchain_bridge/pkg/substrate"
+)
+
+func (bridge *Bridge) handleWithdrawCreated(ctx context.Context, withdraw subpkg.WithdrawCreatedEvent) error {
+	burned, err := bridge.subClient.IsBurnedAlready(types.U64(withdraw.ID))
+	if err != nil {
+		return err
+	}
+
+	if burned {
+		log.Info().Uint64("ID", uint64(withdraw.ID)).Msgf("tx is burned already, skipping...")
+		return errors.New("tx burned already")
+	}
+
+	if err := bridge.wallet.CheckAccount(withdraw.Target); err != nil {
+		log.Info().Uint64("ID", uint64(withdraw.ID)).Msg("tx is an invalid burn transaction, minting on chain again...")
+		mintID := fmt.Sprintf("refund-%d", withdraw.ID)
+		err := bridge.handleMint(big.NewInt(int64(withdraw.Amount)), substrate.AccountID(withdraw.Source), mintID)
+		for err != nil {
+			log.Err(err).Msg("error while minting on chain again for invalid withdraw")
+
+			select {
+			case <-ctx.Done():
+				return err
+			case <-time.After(10 * time.Second):
+				err = bridge.handleMint(big.NewInt(int64(withdraw.Amount)), substrate.AccountID(withdraw.Source), mintID)
+			}
+		}
+		if err != nil {
+			return errors.Wrap(err, "failed to handle mint on chain again for invalid withdraw")
+		}
```

Comment:
This will never trigger. If err is not nil, th eloop above is entered, which only exits either with  a complete return from the functon, or if the handleMint function does not return an error, overriding the old value, thus making sure that err at this point is always nil.